app/src/main/java/com/github/se/studybuddies/viewModels/DirectMessageViewModel.kt

```diff
@@ -3,15 +3,14 @@ package com.github.se.studybuddies.viewModels
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.github.se.studybuddies.data.Chat
-import com.github.se.studybuddies.database.DatabaseConnection
+import com.github.se.studybuddies.database.DbRepository
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
 
-class DirectMessageViewModel(userUid: String = "") : ViewModel() {
-  private val db = DatabaseConnection()
+class DirectMessageViewModel(private val userUid: String = "", private val db: DbRepository) : ViewModel() {
```

Comment:
I am surprised this even compiles! :o I did not know you could define mandatory arguments *after* optional arguments in kotlin

Reply:
Haha same I wasn't sure about it at all at the beggining